subsys/net/lib/coap/coap.c

```diff
@@ -149,6 +149,20 @@ int coap_packet_init(struct coap_packet *cpkt, uint8_t *data, uint16_t max_len,
 	return 0;
 }
 
+int coap_ack_init(struct coap_packet *cpkt, uint8_t *data, uint16_t max_len,
+		  const struct coap_packet *req, uint8_t ver, uint8_t code)
```

Comment:
Actually, we don't need the `ver` parameter in this API, it should be consistent with the request I believe.

Plus, perhaps it's me, but I'd rather reorder to parameters, to provide `req` before `data`, `max_len` and `code`.